File: 1306-minimum-absolute-difference/minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        
        List<List<Integer>> li = new ArrayList<>();
        Arrays.sort(arr);

        int n = arr.length;
        int minV = Integer.MAX_VALUE;

        for(int i=0;i<n-1;i++){
           minV = Math.min(minV,arr[i+1] - arr[i]);    // find diff
        }

        for(int i=0;i<n-1;i++){
            if(minV == arr[i+1]-arr[i]){
                li.add(Arrays.asList(arr[i],arr[i+1]));
            }
        }
        return li;
    }
}

File: 1306-minimum-absolute-difference/minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
          
          vector<vector<int>>v;
          sort(arr.begin(),arr.end());
          int n = arr.size();
           int minDiff = INT_MAX;

          for(int i=0;i<n-1;i++){    // find min Difference
             minDiff = min(minDiff,arr[i+1]-arr[i]);
          }

          for(int i=0;i<n-1;i++){    // Store pairs with minimum difference
             if(minDiff == arr[i+1]-arr[i]){
                v.push_back({arr[i],arr[i+1]});
             }
          }
          return v;
    }
};